### DISCUSSION TITLE
[firebase-storage] enable clearText for localhost

### DISCUSSION THREAD

**Discussion by jarrodonlo** [firebase/flutterfire#7769] [Feature request]
### Connecting to Firebase `Storage` emulator with Android version 9+ will result in error...
```terminal
W/NetworkRequest(11330): error sending network request GET http://10.0.2.2:9199/v0/b/default-bucket/o/cat.jpg
W/NetworkRequest(11330): java.io.IOException: Cleartext HTTP traffic to 10.0.2.2 not permitted
```
As fix, to enable clearText in `debug` mode add the following line to `android/app/src/debug/AndroidManifest.xml`
```xml
    <application android:usesCleartextTraffic="true"/>
```

Or enabling cleartext per host/port per [this article](https://devblogs.microsoft.com/xamarin/cleartext-http-android-network-security/)

**Comment by TAMohamed-Iyed** [comment]
it works, thank you <3

**Comment by at15** [comment]
Thanks to this answer, I didn't realize there are two `AndroidManifest.xml` files... When you run the flutter app directly inside IDE like Intellij, it is using the `src/debug/AndroidManifest.xml` instead of the `src/main/AndroidManifest.xml`. If you only modified the `src/debug/AndroidManifest.xml` you won't see effect for android emulator or real device with debug apk.

Also you can add the network config to allow specific domain https://developer.android.com/privacy-and-security/security-config

In `src/debug/AndroidManifest.xml` i

```xml
<application android:networkSecurityConfig="@xml/network_security_config"/>
```

Create a `src/main/res/xml/network_security_config.xml` and put 

```
<?xml version="1.0" encoding="utf-8"?>
<network-security-config>
    <domain-config cleartextTrafficPermitted="true">
        <domain>10.0.2.2:9199</domain>
    </domain-config>
</network-security-config>
```
